tests/components/hand-controls.test.js

```diff
@@ -0,0 +1,153 @@
+/* global assert, process, setup, suite, test */
+var entityFactory = require('../helpers').entityFactory;
+var controllerComponentName = 'hand-controls';
+
+suite(controllerComponentName, function () {
+  setup(function (done) {
+    var el = this.el = entityFactory();
+    el.setAttribute(controllerComponentName, '');
+    process.nextTick(function () {
+      done();
+    });
+  });
+
+  suite('update', function () {
+    test('oculus-touch-controls and vive-controls injected', function (done) {
+      var el = this.el;
+      var controllerComponent = el.components[controllerComponentName];
+      controllerComponent.update();
+      process.nextTick(function () {
+        assert.ok(el.components['oculus-touch-controls']);
+        assert.ok(el.components['vive-controls']);
+        done();
+      });
+    });
+
+    test('blend-character-model injected', function (done) {
+      var el = this.el;
+      var controllerComponent = el.components[controllerComponentName];
+      controllerComponent.update();
+      process.nextTick(function () {
+        assert.ok(el.components['blend-character-model']);
+        done();
+      });
+    });
+  });
+
+  suite('isOculusTouch', function () {
+    test('true if controller id starts with "Oculus Touch"', function (done) {
+      var el = this.el;
+      var controllerComponent = el.components[controllerComponentName];
+      var trackedControls;
+      el.setAttribute('tracked-controls', '');
+      process.nextTick(function () {
```

Comment:
`nextTick` is apparently needed for DOM updates to take effect before next test portion is run, as mentioned in A-Frame test docs (and experimentally confirmed, i.e. didn't work without)